apps/ejabberd/src/http_upload/mod_http_upload.erl

```diff
@@ -59,40 +76,103 @@ iq_handler(_From, _To, IQ = #iq{type = set, sub_el = SubEl}) ->
 iq_handler(_From, _To, IQ = #iq{type = get, sub_el = Request}) ->
     case parse_request(Request) of
         {Filename, Size, ContentType} ->
-            UTCDateTime = calendar:universal_time(),
-            Token = generate_token(),
-            Opts = module_opts(),
-
-            {PutUrl, GetUrl} = mod_http_upload_backend:create_slot(
-                                 UTCDateTime, Token, Filename, ContentType, Size, Opts),
-
-            compose_iq_reply(IQ, PutUrl, GetUrl);
+            case max_file_size =:= undefined orelse Size =< max_file_size() of
+                true ->
+                    UTCDateTime = calendar:universal_time(),
+                    Token = generate_token(),
+                    Opts = module_opts(),
+
+                    {PutUrl, GetUrl} = mod_http_upload_backend:create_slot(
+                                         UTCDateTime, Token, Filename, ContentType, Size, Opts),
+
+                    compose_iq_reply(IQ, PutUrl, GetUrl);
+
+                false ->
+                    MaxFileSizeBin = integer_to_binary(max_file_size()),
+                    Error0 = ?ERR_NOT_ACCEPTABLE,
+                    Error = Error0#xmlel{
+                              children =
+                                  [
+                                   #xmlel{name = <<"file-too-large">>,
```

Comment:
what about creating separate function which returns this xml error element?
maybe it would calm down Elvis a bit :D